src/libguac/wol.c

```diff
@@ -85,19 +86,31 @@ static ssize_t __guac_wol_send_packet(const char* broadcast_addr,
     wol_dest.sin_family = AF_INET;
     if (inet_pton(AF_INET, broadcast_addr, &(wol_dest.sin_addr)) == 0) {
         wol_dest.sin_family = AF_INET6;
-        if (inet_pton(AF_INET6, broadcast_addr, &(wol_dest.sin_addr)) == 0)
+        if (inet_pton(AF_INET6, broadcast_addr, &(wol_dest.sin_addr)) == 0) {
+            guac_error = GUAC_STATUS_SEE_ERRNO;
+            guac_error_message = "Wake-on-LAN address does not appear to be IPv4 or IPv6";
             return 0;
+        }
     }
     
     /* Set up socket for IPv4 broadcast. */
     if (wol_dest.sin_family == AF_INET) {
         int wol_bcast = 1;
         wol_socket = socket(AF_INET, SOCK_DGRAM, 0);
```

Comment:
Didn't notice this until now, but why `0` for IPv4 and `IPPROTO_UDP` for IPv6? It seems that's the only difference between the `socket()` calls which would prevent using `socket(wol_dest.sin_family, ...)`.

Reply:
Looks like we can probably just specify `0` for both of them and collapse the socket piece into a single declaration, so I'll work on that.  According to `socket()` man page, the socket type (`SOCK_DGRAM`) usually determines the protocol and you can just put `0` for the protocol.  Guessing that will work equally well for IPv6 as IPv4 in this case.

Reply:
This should be a little more efficient, now.